core/store/src/trie/mod.rs

```diff
@@ -1222,6 +1239,58 @@ impl Trie {
         }
     }
 
+    /// Retrieves the value (inlined or reference) for the given key, from the in-memory trie.
+    /// In general, in-memory tries may inline a value if the value is short, but otherwise
+    /// it would defer the storage of the value to the state column. This method will
+    /// return whichever the in-memory trie has.
+    ///
+    /// If an inlined value is returned, this method will charge the corresponding gas
+    /// as if the value were accessed from the trie storage. It will also insert the
+    /// value into the accounting cache, as well as recording the access to the value
+    /// if recording is enabled. In other words, if an inlined value is returned the
+    /// behavior is equivalent to if the trie were used to access the value reference
+    /// and then the reference were used to look up the full value.
+    ///
```

Comment:
Nit: the part about inlining is hard to understand... Can we just say that the charged gas and accounting cache are changed in the same way as if trie was accessed where nothing was inlined? With such, we don't scare reader with internal memtrie / inlining implementation details :)